src/Airship/Internal/Route.hs

```diff
@@ -65,6 +102,42 @@ var t = Route [Var t]
 star :: Route
 star = Route [RestUnbound]
 
+
+-- Routing trie creation algorithm
+-- 1. Store full paths as keys up to first `var`
+-- 2. Take SHA256 digest of the URL portion preceding the
```

Comment:
Yeah the sha256 is probably overkill. I picked it because afaik there's still now known collisions for it and we were already building the dependency. The goal is just to have a unique key into the trie that will not conflict with another request path. I think actually we don't need a hash. I think base64 encoding will work just fine and then the risk of collisions dealing with *lesser* hashes is eliminated. 

I don't completely follow the final comment about the delimiter character. Are you suggesting storing the full route as a key with a delimiter substituted in for the `var` occurrences or something else? That would still require another delimiter character to be able to discern `var` from `star`.